libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3261,8 +3288,12 @@ void TypeChecker::requireLValue(Expression const& _expression, bool _ordinaryAss
 				if (varDecl->isExternalCallableParameter() && dynamic_cast<ReferenceType const*>(identifier->annotation().type))
 					return { 7128_error, "External function arguments of reference type are read-only." };
 
+		fatal = true;
```

Comment:
Why not call `m_errorReporter.fatalTypeError(errorId, _expression.location(), description);` directly here?